src/mxarray.jl

```diff
@@ -406,13 +406,13 @@ end
 # char arrays and string
 
 function mxarray(s::String)
-    utf16string = transcode(UInt16, s)
+    utf16string = transcode(mxChar, s)
```

Comment:
`transcode(UInt16, ...)` is the API to decode it as UTF16, not `transcode(mxChar, ...)`. I don't see how any of the links are relevant.